libgnucash/app-utils/gnc-ui-util.c

```diff
@@ -966,6 +966,13 @@ gnc_find_or_create_equity_account (Account *root,
     g_return_val_if_fail (currency != NULL, NULL);
     g_return_val_if_fail (root != NULL, NULL);
 
+    if (equity_type == EQUITY_OPENING_BALANCE)
+        account = gnc_account_lookup_by_opening_balance (root, currency);
+
+    if (account &&
+            gnc_commodity_equiv (currency, xaccAccountGetCommodity (account)))
+        return account;
```

Comment:
You need to initialize account to NULL at declaration to prevent a crash in the EQUITY_RETAINED_EARNINGS case. But it would be better to `if (account) return account` in the previous if-block. There's no need to test the commodity match, `gnc_account_lookup_by_opening_balance` already did that.